### PR TITLE
Gertsonderby map list

### PR DESCRIPTION
**Pull request by erikmueller**
@gertsonderby I took the freedom to shuffle your commits a bit. 

Mostly the following
* changed some wording
* use `new Map()` instead of `Immutable.Map()`
* `var` = `const`
* use `Immutable.fromJS()` for creating nested Immutables (instead of `new List(new Map({}))`)

Since it was your PR (and work) just tell me if this is ok and I will merge your commits.

### CODE REVIEW

**Comment by gertsonderby** [comment]
I generally prefer factory functions over `new` constructs, especially since immutable objects are a functional language construct. But this is your lib, as long as it works, it's a preference matter.

The main reason I used `var`, rather than `const`, was for compatibility. If you have a way around that for not fully ES2015 platforms, I'm more than happy using the new keywords.

And lastly, I agree with using `fromJS()`, I should have done that.

TL;DR: Go ahead. :)
